### ISSUE TITLE
Solidity function which take `bytes` as an argument doesn't work properly

### ISSUE THREAD

**Issue by sdesmond46** [ethereum-ts/TypeChain#41]
If you have a solidity contract which takes an argument of type `bytes`, the typechain conversions will change the data you're sending in. The current definitions expect `BigNumber[]` to be the passed in value for `bytes`. When we convert that BigNumber array, it gets split out into a totally different format than what we input. For Examples:

```solidity
contract TypeChainBugDemo {
  // Calling this returns an unexpected value
  function callWithBytes(bytes bytesArg) public returns (uint256){
    return bytesArg.length;
  }
}
```

This generates the following typescript
```typescript
public callWithBytes(bytesArg: BigNumber[]): Promise<BigNumber> {
  return promisify(this.rawWeb3Contract.callWithBytes, [bytesArg.map(val => val.toString())]);
}
```

As a caller, I assumed if I wanted to pass a bytes string of length 2, represented by the string `"0xA20F"`, I would pass `[new BigNumber('0xA2'), new BigNumber('0x0F')]`. Unfortunately, when this gets converted by typechain, it will change to ["162", "15"]. When that gets passed into web3, it actually cocatenates these strings, then gets their ASCII values for each character. Ultimately, the 2 byte input array gets expanded into a totally different 5 byte array.

I think the right fix is to have bytes always take in a `string` for TypeChain. If you want the string to be interpreted as hex, it's just prepended with 0x and web3 takes care of things. Otherwise, web3 converts to utf-8 and passes in the byte stream for the string.

**Comment by krzkaczor** [comment]
Closed by #42. Finally decided to use strings to pass bytes arrays.

**Comment by eswarasai** [comment]
@krzkaczor -- Is this issue also fixed for `bytes32`? I'm using Typechain `0.2.1` and it's converting `bytes32` to `BigNumber` instead of a `string`. Any resolution for this? Thanks!
